### ISSUE TITLE
Errror: EMAXBUFFER when trying to upload via sails-skippers3

### ISSUE THREAD

**Issue by ebbnormal** [balderdashy/skipper#121]
So I have a bunch of form data that I am submitting with a post request. Some of the fields are files, So created a service which, given a file, pushes it to S3. I call this service in my controller. 

Here is my Service: 

    var UploadFileToS3 = {
	uploadS3: function uploadService(file, filename, key, bucket, secret, cb){
        	    file.upload({
				adapter: require('skipper-s3'),
		   	key: key,
		   	secret: secret,
		   	bucket: bucket,
		   	saveAs: filename
	    }, function whenDone(err, uploadedFiles) {
			  if (err) {
			   	return cb(err);
			  }
			  if (uploadedFiles.length === 0) {
			  	return cb({success:false, error:err});
			  }
	  });
	}
       };
      module.exports = UploadFileToS3;

Here is how it is called in my controller method: 

     createFounderApplication: function(req, res, next){
	  console.log("in create Founder Application");
	  console.log(req.body);
	  console.log(req.files);
	  var paramNames = ['avatartwo']; //TODO put all files param names here
      async.forEach(paramNames, function (file, callback){
        UploadFileToS3.uploadS3(req.file(file), 
        	                  req.file(file).name, 
        	                  <mykey>,
        	                  'thunderboltalpha', 
        	                 <secret>,
        	                  callback);
        }, function (err, res){
    	   if (err) return next(err); 
           res.json({ success:true, message: "uploaded files"});
        });
	},

My issue is that when I call the POST method which hits `createFounderApplication` I get the following error: 

     error: Sending 500 ("Server Error") response: 
     { success: false, error: null }

    events.js:72
        throw er; // Unhandled 'error' event
              ^
    Error: EMAXBUFFER: An Upstream (`NOOP_avatartwo`) timed out before it was plugged into a     receiver. It was still unused after waiting 4500ms. You can configure this timeout by changing the `maxTimeToBuffer` option.
    at null.<anonymous>   (/Users/fd/code/fgf/myproj/node_modules/skipper/standalone/Upstream/Upstream.js:86:15)
    at Timer.listOnTimeout [as ontimeout] (timers.js:110:15)

**Comment by ebbnormal** [comment]
Alternatively, if there is a way I can upload each file to S3 in the controller without going to the Service, and that doesn't break it, I suppose that is fine, although that would make my controller undesirably overweight.

**Comment by sailsbot** [comment]
Thanks for posting, @ebbnormal.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
